plugins/hydra_ray_launcher/hydra_plugins/hydra_ray_launcher/ray_aws_launcher.py

```diff
@@ -0,0 +1,218 @@
+    JOB_RETURN_PICKLE,
+    JOB_SPEC_PICKLE,
+)
+
+log = logging.getLogger(__name__)
+
+
+class RayAWSLauncher(Launcher):
+    def __init__(
+        self,
+        ray_init_cfg: DictConfig,
+        ray_remote_cfg: DictConfig,
+        ray_cluster_cfg: DictConfig,
+        stop_cluster: bool,
+    ) -> None:
+        self.ray_init_cfg = ray_init_cfg
+        self.ray_remote_cfg = ray_remote_cfg
+        self.ray_cluster_cfg = ray_cluster_cfg
+        self.docker_enabled = self.ray_cluster_cfg.docker.image != ""
+        self.stop_cluster = stop_cluster
+        self.config: Optional[DictConfig] = None
+        self.config_loader: Optional[ConfigLoader] = None
+        self.task_function: Optional[TaskFunction] = None
+        self.ray_yaml_path: Optional[str] = None
+
+    def setup(
+        self,
+        config: DictConfig,
+        config_loader: ConfigLoader,
+        task_function: TaskFunction,
+    ) -> None:
+        self.config = config
+        self.config_loader = config_loader
+        self.task_function = task_function
+
+    def launch(
+        self, job_overrides: Sequence[Sequence[str]], initial_job_idx: int
+    ) -> Sequence[JobReturn]:
+        setup_globals()
+        assert self.config is not None
+        assert self.config_loader is not None
+        assert self.task_function is not None
+
+        configure_log(self.config.hydra.hydra_logging, self.config.hydra.verbose)
+        sweep_dir = Path(str(self.config.hydra.sweep.dir))
+        sweep_dir.mkdir(parents=True, exist_ok=True)
+        log.info(f"Ray Launcher is launching {len(job_overrides)} jobs")
+        log.info(f"Sweep output dir : {sweep_dir}")
+
+        with tempfile.TemporaryDirectory() as local_tmp_dir:
+            sweep_configs = []
+            for idx, overrides in enumerate(job_overrides):
+                idx = initial_job_idx + idx
+                ostr = " ".join(filter_overrides(overrides))
+                log.info(f"\t#{idx} : {ostr}")
+                sweep_config = self.config_loader.load_sweep_config(
+                    self.config, list(overrides)
+                )
+                with open_dict(sweep_config):
+                    # job.id will be set on the EC2 instance before running the job.
+                    sweep_config.hydra.job.num = idx
+
+                sweep_configs.append(sweep_config)
+
+            self._pickle_jobs(
+                tmp_dir=local_tmp_dir,
+                sweep_configs=sweep_configs,
+                task_function=self.task_function,
+                singleton_state=Singleton.get_state(),
+            )
+
+            with tempfile.NamedTemporaryFile(suffix=".yaml", delete=False) as f:
+                with open(f.name, "w") as file:
+                    print(
+                        OmegaConf.create(
+                            OmegaConf.to_container(
+                                self.ray_cluster_cfg, enum_to_str=True
+                            )
+                        ).pretty(),
```

Comment:
File a feature request for OmegaConf
s pretty() to support enum_to_str and put a TODO here referencing that issue.